libqpdf/qpdf/Pl_TextIfElse.hh

```diff
@@ -0,0 +1,30 @@
+#ifndef PL_TEXTIFELSE_HH
+#define PL_TEXTIFELSE_HH
+
+#include <qpdf/Pipeline.hh>
+
+// This pipeline has no `next` associated with the base class. Instead, it picks which of two
+// candidate `next`s to call based on its detection of text with a bias toward text consisting
+// of characters <= 0x7FF. I had considered writing a general IfElse pipeline that takes a closure,
+// but I decided that's premature generalization.
+class Pl_TextIfElse: public Pipeline
+{
+  public:
+    Pl_TextIfElse(char const* identifier, Pipeline* next_if_text, Pipeline* next_if_binary);
+    ~Pl_TextIfElse() override;
```

Comment:
I would be less general and write this in the knowledge that next_is_binary is the decrypt pipeline, and change the parameter to a unique_ptr<Pipeline>.  This allows us to move the decrypt pipeline into Pl_TextIfElse and setting the decrypt pipeline to  Pl_TextIfElse